src/Storages/MutationCommands.cpp

```diff
@@ -182,4 +187,46 @@ void MutationCommands::readText(ReadBuffer & in)
     }
 }
 
+void MutationCommands::writeSettings(WriteBuffer & out) const
+{
+    for (auto it = settings_changes.begin(); it != settings_changes.end(); ++it)
+    {
+        if (it != settings_changes.begin())
+            out << ", ";
+        formatSettingName(it->name, out);
+        out << " = ";
+        writeEscapedString(applyVisitor(FieldVisitorToString(), it->value), out);
+    }
+}
+
+void MutationCommands::readSettings(ReadBuffer & in)
+{
+    String settings_str;
+    readEscapedString(settings_str, in);
+    ParserSetQuery p_settings(true);
+    ASTPtr ast = parseQuery(
+        p_settings, settings_str.data(), settings_str.data() + settings_str.length(), "mutation settings", 0, DBMS_DEFAULT_MAX_PARSER_DEPTH);
+    settings_changes = std::move(ast->as<ASTSetQuery>()->changes);
```

Comment:
Consider moving this code to a new method of `SettingChange` as it's related to `SettingChange`, not to `MutationCommands` (and the same method in needed for `DDLLogEntry`)

Reply:
That would introduce a dependency between `src/Common` and `src/Parsers`, which is probably not okay?

Reply:
Yep, it's not okay :(